Dialog to reorder station list, station names now contain hidden list index value

pelya opened the pull request:
Demonstration: https://www.youtube.com/watch?v=r4xsxcBw7fc

y20k commented:
Hi there @pelya,

Thanks for offering this new feature!! Although I would have liked to have a discussion (see [CONTRIBUTING.md](https://github.com/y20k/transistor/blob/master/CONTRIBUTING.md)), because I am always afraid that a contributor invests too much work in code I cannot accept.

First a question: 
- in your solution you add a number to the station name to implement the custom sort order, right? (and you strip the number before displaying the name) -> I like this. A very lean solution. 👍 

My problems with the implementation in UI: 
- I do not want to use external libraries in Transistor - meaning non-Google ones
- A reorder feature should in my opnion be implementend natively on the recyclerview list using [ItemTouchHelper](https://developer.android.com/reference/android/support/v7/widget/helper/ItemTouchHelper) - I implemented a "swipe to delete" feature over at my other (unfinished) app [Escapepods](https://github.com/y20k/escapepods) using this nice [ItemTouchHelper-Tutorial](https://medium.com/@kitek/recyclerview-swipe-to-delete-easier-than-you-thought-cff67ff5e5f6)

Solutions:
a) I can take your backend-code and implement the UI using ItemTouchHelper
b) you give it another round of hacking and do it yourself

Hope you are not disappointed. But I cannot just use the pull request as it is. Sorry.

Best wishes

pelya commented:
I was actually trying to fix the crash, which happens on Play Store version
when you stop playback and do other things for a while so system will stop
inactive apps, then you open Transistor and play the same station and it
crashes. You should have some crash logs in Play Store dev console, I've
submitted like 10 of them.

But I could not reproduce the crash on the version compiled from Git, and I
had some free time, so I made this patch instead.

I could not find any good answer for how to create a sortable list
natively, every answer seems to be some kind of 3rd party library. Every
app does it differently, here's how it's done in Flud, which is honestly
terrible:  https://i.imgur.com/LWOZJTY.png

Anyway, I'll try to think how to do it without 3rd party library, but this
probably won't happen until next month.

It does not bother me if you keep this patch in limbo for a while, it
solves one minor annoyance, but it's not that much important of a feature.


On Fri, Nov 16, 2018, 12:00 y20k <notifications@github.com wrote:

> Hi there @pelya <https://github.com/pelya>,
>
> Thanks for offering this new feature!! Although I would have liked to have
> a discussion (see CONTRIBUTING.md
> <https://github.com/y20k/transistor/blob/master/CONTRIBUTING.md>),
> because I am always afraid that a contributor invests too much work in code
> I cannot accept.
>
> First a question:
>
>    - in your solution you add a number to the station name to implement
>    the custom sort order, right? (and you strip the number before displaying
>    the name) -> I like this. A very lean solution. 👍
>
> My problems with the implementation in UI:
>
>    - I do not want to use external libraries in Transistor - meaning
>    non-Google ones
>    - A reorder feature should in my opnion be implementend natively on
>    the recyclerview list using ItemTouchHelper
>    <https://developer.android.com/reference/android/support/v7/widget/helper/ItemTouchHelper>
>    - I implemented a "swipe to delete" feature over at my other (unfinished)
>    app Escapepods <https://github.com/y20k/escapepods> using this nice
>    ItemTouchHelper-Tutorial
>    <https://medium.com/@kitek/recyclerview-swipe-to-delete-easier-than-you-thought-cff67ff5e5f6>
>
> Solutions:
> a) I can take your backend-code and implement the UI using ItemTouchHelper
> b) you give it another round of hacking and do it yourself
>
> Hope you are not disappointed. But I cannot just use the pull request as
> it is. Sorry.
>
> Best wishes
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/y20k/transistor/pull/202#issuecomment-439343204>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAJewE-uzGGnzo8wzyBde_MLKy5sv0coks5uvozbgaJpZM4YilEi>
> .
>

On Nov 16, 2018 12:00 PM, "y20k" <notifications@github.com> wrote:

Hi there @pelya <https://github.com/pelya>,

Thanks for offering this new feature!! Although I would have liked to have
a discussion (see CONTRIBUTING.md
<https://github.com/y20k/transistor/blob/master/CONTRIBUTING.md>), because
I am always afraid that a contributor invests too much work in code I
cannot accept.

First a question:

   - in your solution you add a number to the station name to implement the
   custom sort order, right? (and you strip the number before displaying the
   name) -> I like this. A very lean solution. 👍

My problems with the implementation in UI:

   - I do not want to use external libraries in Transistor - meaning
   non-Google ones
   - A reorder feature should in my opnion be implementend natively on the
   recyclerview list using ItemTouchHelper
   <https://developer.android.com/reference/android/support/v7/widget/helper/ItemTouchHelper>
   - I implemented a "swipe to delete" feature over at my other (unfinished)
   app Escapepods <https://github.com/y20k/escapepods> using this nice
   ItemTouchHelper-Tutorial
   <https://medium.com/@kitek/recyclerview-swipe-to-delete-easier-than-you-thought-cff67ff5e5f6>

Solutions:
a) I can take your backend-code and implement the UI using ItemTouchHelper
b) you give it another round of hacking and do it yourself

Hope you are not disappointed. But I cannot just use the pull request as it
is. Sorry.

Best wishes

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/y20k/transistor/pull/202#issuecomment-439343204>, or mute
the thread
<https://github.com/notifications/unsubscribe-auth/AAJewE-uzGGnzo8wzyBde_MLKy5sv0coks5uvozbgaJpZM4YilEi>
.

y20k commented:
If you find any way to reproduce the seemingly random crash, please drop me note. It happened to me too. Thx

y20k commented:
Oh. I found a way to reproduce the crash. I will open a new issue here and describe what's going on there.

y20k commented:
FYI @pelya the seemingly random crash is now issue #199
